pycm/pycm_curve.py

```diff
@@ -0,0 +1,123 @@
+        :type classes: list
+        :param thresholds: thresholds list
+        :type thresholds: list
+        :param sample_weight: sample weights list
+        :type sample_weight: list
+        """
+        self.data = {}
+        self.thresholds = []
+        __curve_validation__(actual_vector, prob_vector)
+        self.actual_vector = actual_vector
+        self.prob_vector = prob_vector
+        __curve_classes_handler__(self, classes)
+        __curve_thresholds_handler__(self, thresholds)
+        for c_index , c in enumerate(self.classes):
+            data_temp = {item:[] for item in CURVE_PARAMS}
+            for t in self.thresholds:
+                lambda_fun = lambda x: threshold_func(x, c_index, classes, t)
+                cm = ConfusionMatrix(actual_vector=self.actual_vector, predict_vector=self.prob_vector, threshold=lambda_fun, sample_weight=sample_weight)
+                for item in CURVE_PARAMS:
+                    data_temp[item].append(getattr(cm, item)[c])
+            self.data[c] = data_temp
+        self.plot_x_axis = None
+        self.plot_y_axis = None
+
+    def area(self):
+        pass
+
+    def plot(self):
+        pass
+
+
+def __curve_validation__(actual_vector, prob_vector):
+    """
+    Curve input validation.
+
+    :param actual_vector: actual vector
+    :type actual_vector: python list or numpy array of any stringable objects
+    :param prob_vector: vector of probabilities
+    :type prob_vector: python list or numpy array of any stringable objects
+    :return: None
+    """
+    if not isinstance(actual_vector, (list, numpy.ndarray)) or not isinstance(prob_vector, (list, numpy.ndarray)):
+        raise pycmCurveError(VECTOR_TYPE_ERROR)
+    if len(actual_vector) != len(prob_vector):
+        raise pycmCurveError(VECTOR_SIZE_ERROR)
+    if len(set(map(len, prob_vector))) != 1:
+        raise pycmCurveError(PROBABILITY_SIZE_ERROR)
+    for item in prob_vector:
+        if not all(map(isfloat, item)):
+            raise pycmCurveError(PROBABILITY_TYPE_ERROR)
+        if sum(item) != 1:
+            raise pycmCurveError(PROBABILITY_SUM_ERROR)
+
+
+def __curve_classes_handler__(curve, classes):
+    """
+    Handle conditions for curve classes.
+
+    :param curve: curve
+    :type curve: pycm.Curve object
+    :param classes: ordered labels of classes
+    :type classes: list
+    :return: None
+    """
+    if not isinstance(classes, list):
+        raise pycmCurveError(CLASSES_TYPE_ERROR)
+    if len(set(classes)) != len(classes):
+        raise pycmCurveError(VECTOR_UNIQUE_CLASS_ERROR)
+    if set(classes) != set(curve.actual_vector):
+        raise pycmCurveError(CLASSES_MATCH_ERROR)
+    if len(set(classes)) < 2:
+        raise pycmCurveError(CLASS_NUMBER_ERROR)
+    curve.classes = classes
+    if len(set(map(type, curve.actual_vector))) > 1:
+        curve.classes = list(map(str, curve.classes))
+
+
+def __curve_thresholds_handler__(curve, thresholds):
```

Comment:
Why did you use double dunders?
How about `_curve_thresholds_handler` since it is an inner-use function?